src/abr/pensieve.cc

```diff
@@ -0,0 +1,105 @@
+#include "pensieve.hh"
+#include "ws_client.hh"
+
+using namespace std;
+
+//TODO: Reference already existing put_field()
+string put_field2(const uint16_t n)
+{
+  const uint16_t network_order = htobe16(n);
+  return string(reinterpret_cast<const char *>(&network_order),
+                sizeof(network_order));
+}
+
+//TODO: Reference already existing get_uint16()
+uint16_t get_uint16_2(const char * data)
+{
+  return be16toh(*reinterpret_cast<const uint16_t *>(data));
+}
+
+Pensieve::Pensieve(const WebSocketClient & client,
+         const string & abr_name, const YAML::Node & abr_config)
+  : ABRAlgo(client, abr_name)
+{
+
+    cout << abr_config << endl; // TODO: Actually use abr_config or remove it
+    // Config right now is done via constants in the Pensieve file, I am tempted
+    // to leave it that way unless we think we will actually use different
+    // configs for different people/channels.
+}
+
+void Pensieve::video_chunk_acked(const VideoFormat & format,
+                            const double ssim,
+                            const unsigned int size, //total length in bytes of chunk
+                            const uint64_t trans_time) //ms
+{
+    const auto & channel = client_.channel();
+    const auto & vformats = channel->vformats();
+    size_t vformats_cnt = vformats.size();
+
+    uint64_t next_vts = client_.next_vts().value();
+    const auto & data_map = channel->vdata(next_vts);
+    vector<pair<double, size_t>> next_chunk_sizes; //Store pairs of chunk size, corresponding vf index
+    for (size_t i = 0; i < vformats_cnt; i++) {
+      const auto & vf = vformats[i];
+
+      size_t chunk_size = get<1>(data_map.at(vf));
+      double chunk_size_mb = (double)chunk_size; // Bytes
```

Comment:
Why not merge this type conversion into the next line? Or simply reply on the implicit conversion?